### PR TITLE
fix: add codable conformance to byte buffer plus extension

### PR DESCRIPTION
**Pull request by kneekey23**
*Description of changes:* This PR adds needed extensions to use `ByteBuffer` in streaming in the sdk

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift [hunk 1]
```diff
@@ -178,6 +191,10 @@ public class ByteBuffer {
             : Double(bitPattern: result.bigEndian)
     }
 
+    public func toByteArray() -> [UInt8] {
+        return array
+    }
+
```
Comment (wooj2) [inline Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift:197]: per convo: if we don't need this, lets remove it

File: Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift [hunk 2]
```diff
@@ -49,6 +49,19 @@ public class ByteBuffer {
         return self
     }
 
+    public func put(_ value: ByteBuffer) {
+        array.append(contentsOf: value.array)
+    }
+
+    public func put(_ value: ByteBuffer, offset: UInt = 0, maxBytes: UInt?) {
```
Comment (wooj2) [inline Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift:56]: * if we default offset to 0, then it should come after variables that do not have defaults
* investigate combining both put functions to single function.

Comment (justinboswell) [reply Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift:56]: +1, these are the same function if `offset` and `maxBytes` have defaults

**Review by justinboswell — APPROVED** [review]

File: Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift (hunk 2 above)
Comment (kneekey23) [reply Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift:56]: merging into one function

**Review by justinboswell — APPROVED** [review]

**Pull request merged**